lstchain/reco/disp.py

```diff
@@ -36,7 +37,16 @@ def disp(cog_x, cog_y, src_x, src_y):
     """
     disp_dx = src_x - cog_x
     disp_dy = src_y - cog_y
-    disp_norm = np.sqrt(disp_dx**2 + disp_dy**2)
+    
+    disp_norm = disp_dx * np.cos(hillas_psi) + disp_dy * np.sin(hillas.psi)
+    disp_sign = np.sign(disp_norm)
+    disp_norm = abs(disp_norm)
```

Comment:
Actually, @maxnoe, everyone: is there any reason not to simply leave the sign in "disp_norm", and do the reconstruction with RF of the signed quantity? We would save one RF, and I see no reason why it should be worse than doing norm & sign separately.

Reply:
We can give it a try. I think I tried it out for FACT and got slightly worse performance, but not sure anymore. Classification plus regression is not using the same loss function as only regression... 

The separate prediction mainly comes from the history of this algorithm I'd say

Reply:
One thing that came to mind is that a single random forest predicting disp with sign will behave differently in cases where the sign prediction is difficult/impossible. 

The two model version will guess a sign and predict an okish disp value, leading to the "ghost events" on the other side of the true source, very far away.

The single model version will guess values close to 0, leading to ghost events close to the center of gravity, so much closer to the true source position.

This seems like it could be worse for point sources?

Have not tested it though

Reply:
if we go for a reduction in the number of parameters fed to the RF, this may imply a major release instead of a bugfix one. Additionally, since Max is saying that the performance may be slightly worse, I would go for the solving of this bug and opening an issue to keep this improvement in mind, test it after its implementation to see if the performance is the same, and propose it in a new PR to be released in a future major release.

Reply:
Yes, I did not intend it to be part of this fix, of course! Max's argument kind of convinces me, so let's park that for now.

Reply:
great!